bottlerocket/testsys/src/run_aws_ecs.rs

```diff
@@ -350,6 +360,27 @@ impl RunAwsEcs {
         secrets: Option<BTreeMap<String, SecretName>>,
         cluster_resource_name: &str,
     ) -> Result<Resource> {
+        let user_data = if let Some(path) = &self.custom_user_data {
+            let parse_result = read_to_string(path.clone());
+            match parse_result {
+                Ok(data) => match self.custom_user_data_mode.as_deref() {
+                    Some("merge") => Ok(Some(CustomUserData::Merge {
+                        encoded_userdata: data,
+                    })),
+                    Some("replace") => Ok(Some(CustomUserData::Replace {
+                        encoded_userdata: data,
+                    })),
+                    _ => Err(Error::InvalidArguments {
+                        why: "invalid custom userdata mode".to_string(),
+                    }),
```

Comment:
This is the error we should avoid. Especially since the "invalid custom userdata mode" might be `Option::None`.

If you add a "tagging" enum like I requested above, which I called `CustomerUserdataMode`, you could:

```rust
impl CustomUserData {
    fn new(x: CustomUserDataConfig) -> Self {
       match x.mode {
           /* ... */
       }
   }
}
```